Add support to getting statistics for loadbalancer and listener

velp opened the pull request:
In this PR:

1. Add GetStats method for a loadbalancer. See API guide: https://developer.openstack.org/api-ref/load-balancer/v2/#get-load-balancer-statistics. Data structure defined in https://github.com/openstack/octavia/blob/master/octavia/api/v2/types/load_balancer.py#L176
2. Add GetStats method for a listener. See API guide: https://developer.openstack.org/api-ref/load-balancer/v2/#get-listener-statistics. Data structure defined in https://github.com/openstack/octavia/blob/master/octavia/api/v2/types/listener.py#L230

For #1189

jtopjian commented:
@velp Thank you for working on this. 

A few initial notes:


> In this PR:

There are a lot of changes happening in this Pull Request and it looks like they can all be implemented independently of each other. Would it be possible to break these up into 3 different PRs?

> Add GetStats method for a loadbalancer

Are you able to reference where in the LBaaS code the data structure is defined? Meaning, can you prove `active_connections` is a valid field? 

> Add GetStats method for a listener

Same here

We have a [Contributor Tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial) that covers these items and might be helpful.

Please let me know if you have any questions. Thank you again.

velp commented:
@jtopjian I updated this PR, deleted two parts (#1195, #1196).

jtopjian commented on openstack/loadbalancer/v2/listeners/results.go:
```diff
@@ -60,6 +60,23 @@ type Listener struct {
 	ProvisioningStatus string `json:"provisioning_status"`
 }
 
+type StatsTree struct {
```
Let's call this `Stats`

velp replied:
Fix

jtopjian commented on openstack/loadbalancer/v2/listeners/results.go:
```diff
@@ -133,3 +150,19 @@ type UpdateResult struct {
+
+// GetStatsResult represents the result of a GetStats operation.
+// Call its Extract method to interpret it as a StatsTree.
+type GetStatsResult struct {
```
Let's call this `StatsResult`

velp replied:
Fix

jtopjian commented on openstack/loadbalancer/v2/loadbalancers/results.go:
```diff
@@ -58,6 +58,23 @@ type StatusTree struct {
 	Loadbalancer *LoadBalancer `json:"loadbalancer"`
 }
 
+type StatsTree struct {
```
Let's call this `Stats`

velp replied:
Fix

jtopjian commented on openstack/loadbalancer/v2/loadbalancers/results.go:
```diff
@@ -124,6 +141,22 @@ func (r GetStatusesResult) Extract() (*StatusTree, error) {
 
+// GetStatsResult represents the result of a GetStats operation.
+// Call its Extract method to interpret it as a StatsTree.
+type GetStatsResult struct {
```
Let's call this `StatsResult`

velp replied:
Fix

velp commented:
@jtopjian thank you for the review. 
> In addition, could you update the doc.go file in both listeners and loadbalancers to add an example of using this?

Added.

> If you're able to, would you also be able to update the acceptance test at acceptance/openstack/loadbalancer/v2/loadbalancers_test.go with these new functions?

Added.
